### DISCUSSION TITLE
How to add software once the vhd is created?

### DISCUSSION THREAD

**Discussion by danielmigueles** [actions/virtual-environments#8369] [General questions]
Once I have the vhd, I would like to add some linux apps into the agent like ansible-galaxy collection. How should I do it?

**Comment by vpolikarpov-akvelon** [comment]
Hi @danielmigueles. There are several options, and the choice depends on your specific goals. You can modify the image by creating a new virtual machine (VM) based on it, make the necessary changes, and then save the updated image. This can be done either manually or using tools like Packer or Azure VM Image Builder. Alternatively, you can choose to install the required software directly on the VMs after launching them. These are just examples, and there may be other approaches available.
